### ISSUE TITLE
Composition fails in Firefox

### ISSUE THREAD

**Issue by darobin** [ianstormtaylor/slate#211]
Steps to reproduce, on a Mac keyboard attempt to produce an "à" character by composing ` and a (Alt-`, a). This can be reproduced with other combinations.

This works in Chrome because the sequence of events is:

```
compositionstart 
compositionupdate ` (ignored by Slate)
keydown à
compositionupdate à (ignored by Slate)
compositionend à
```

In Firefox there is no `keydown` event:

```
compositionstart 
compositionupdate ` (ignored by Slate)
compositionupdate à (ignored by Slate)
compositionend à
```

WebKit's handling is closer to Chrome's (it uses `keydown`) but is yet different in that there is no `compositionupdate` with the fully-composed character: it skips straight to `compositionend`. Fun!

There are several ways in which this could be handled. One would be to just lift what Draft.js does in `src/component/handlers/composition/DraftEditorCompositionHandler.js`. It's a bit brutal but looking through the source they do take care of some weird corner cases (weirder than this one).

Another option would be to use `compositionupdate` events. This could be done across the board (I'm travelling and can't test exact support, but at first pass it seems decent). I looked into synthesizing a `keydown` event from `compositionupdate` but it doesn't seem possible (and would likely be tricky in some languages anyway). This involves flat out ignoring `keydown` while `isComposing`.

Finally there's the option of trying to unify this directly at the React level. This would be great but I can't think of doing that in a way that wouldn't break backwards compatibility.

**Comment by ianstormtaylor** [comment]
Actually, I think this is related to the same bug as https://github.com/ianstormtaylor/slate/issues/210, because the editor is accidentally being re-rendered after every input! Fixed in `0.10.2`, but if you can still reproduce let me know!

**Comment by ianstormtaylor** [comment]
Hold up, not right, fixing.

**Comment by ianstormtaylor** [comment]
Okay fixed in `0.10.4`!

**Comment by darobin** [comment]
Fix confirmed — thanks!
